File: problem_0276/LeetCode2.java
class Solution {
    public int numWays(int n, int k) {
        if (n == 0) {
            return 0;
        }
        if (n == 1) {
            return k;
        }
        int last = k * k;
        int secondLast = k;
        for (int i = 3; i <= n; i++) {
            int tmp = (last + secondLast) * (k - 1);
            secondLast = last;
            last = tmp;
        }
        return last;
    }
}

File: algorithms/problem_0276/leetcode2.py
class Solution:
    def numWays(self, n: int, k: int) -> int:
        if n == 0:
            return 0
        if n == 1:
            return k
        second_last, last = k, k * k
        for i in range(n - 2):
            second_last, last = last, (second_last + last) * (k - 1)
        return last
